### PR TITLE
Remove Dart_GetField from Canvas interface

### PR DESCRIPTION
**Pull request by abarth**
These Dart_GetField calls show up on traces. This patch removes almost all of
them from the Canvas interface. We still have a few in drawAltas and
drawVertices, which will be removed in a later patch.

### CODE REVIEW

**Comment by abarth** [comment]
@eseidelGoogle

File: sky/engine/core/dart/painting.dart [hunk 1]
```diff
@@ -469,6 +574,7 @@ class Canvas extends NativeFieldWrapperClass2 {
     }
     _drawVertices(vertexMode.index, vertices, textureCoordinates, colors, transferMode, indicies, paint);
   }
+  // TODO(abarth): Convert to primatives.
```
Comment (Hixie) [inline sky/engine/core/dart/painting.dart]: "primitives"

**Comment by Hixie** [comment]
LGTM

**Comment by j9brown** [comment]
Interesting.  Fwiw, if you have any JNI code doing similar things you
should be aware that the same kind of optimization is worthwhile there.
Crossing VM boundaries tends to be expensive.

**Comment by abarth** [comment]
We have very little JNI and instead use Mojo for most of the Java communication.

**Comment by j9brown** [comment]
You might want to take a peek at the Mojo JNI code...

**Comment by abarth** [comment]
In our moment of triumph?

**Comment by j9brown** [comment]
Haha!  Well, at least you're not pushing canvas calls over that channel.  :)

**Pull request merged**